src/ooni/web_connectivity.cpp

```diff
@@ -0,0 +1,677 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include <measurement_kit/http.hpp>
+#include <measurement_kit/ooni.hpp>
+
+#include <measurement_kit/net/transport.hpp>
+
+#include "src/common/utils.hpp"
+#include "src/ooni/utils.hpp"
+
+#include <set>
+#include <regex>
+#include <cctype>
+#include <string>
+#include <iostream>
+#include <algorithm>
+#include <unistd.h>
+
+#define BODY_PROPORTION_FACTOR 0.7
+
+namespace mk {
+namespace ooni {
+
+using namespace mk::report;
+
+// These are very common server headers that we don't consider when checking
+// between control and experiment.
+static const std::set<std::string> COMMON_SERVER_HEADERS = {
+  "date",
+  "content-type",
+  "server",
+  "cache-control",
+  "vary",
+  "set-cookie",
+  "location",
+  "expires",
+  "x-powered-by",
+  "content-encoding",
+  "last-modified",
+  "accept-ranges",
+  "pragma",
+  "x-frame-options",
+  "etag",
+  "x-content-type-options",
+  "age",
+  "via",
+  "p3p",
+  "x-xss-protection",
+  "content-language",
+  "cf-ray",
+  "strict-transport-security",
+  "link",
+  "x-varnish"
+};
+
+
+static const std::map<std::string, std::string> COMMON_CLIENT_HEADERS = {
```

Comment:
Here you'd better use `http::Headers`, which soon will implement case insensitive search.

**TODO for me**: implement case insensitive search.